app/Console/Commands/SendConfirmations.php

```diff
@@ -0,0 +1,135 @@
+        'my-museum-tour.show' => MyMuseumTour::class,
+    ];
+
+    /**
+     * Execute the console command.
+     */
+    public function handle()
+    {
+        foreach (self::$models as $route => $modelClass) {
+            $models = $modelClass::notSent()->get();
+
+            foreach ($models as $model) {
+                try {
+                    $this->sendConfirmation($model, $route);
+                } catch (\Exception $exception) {
+                    $this->error($exception->getMessage());
+                    return 1;
+                }
+            }
+        }
+    }
+
+    protected function sendConfirmation($model, $route = null)
+    {
+        $route = empty($route) ? self::route($model) : $route;
+        if (empty($route)) {
+            return false;
+        }
+
+        set_time_limit(0);
+
+        $this->send($model);
+        $model->confirmation_sent = true;
+        $model->save();
+
+        $class = class_basename($model);
+        $this->info("Confirmation sent for {$class} ID {$model->id}");
+    }
+
+    protected function path($model, $route): string
+    {
+        return route($route, ['id' => $model->id], false);
+    }
+
+    protected static function route($model): string
+    {
+        return array_search(get_class($model), self::$models);
+    }
+
+    protected function send($model)
+    {
+        $to = $model->creator_email;
+        $apiCode = config('sendgrid.apikey');
+        $sender = config('sendgrid.email');
+        $templateId = config('sendgrid.template_id');
+
+        $baseUrl = config('aic.protocol') . '://' . config('app.url');
+        $tourUrl = $baseUrl . route('my-museum-tour.show', [ 'id' => $model->id ], false);
+
+        $dynamicContent = [
+            'personalizations' => [
+                [
+                    'to' => [
+                        [
+                            'email' => $to
+                        ]
+                    ],
+                    'dynamic_template_data' => [
+                        'name' => $model->tour_json['creatorName'],
+                        'tour_name' => $model->tour_json['title'],
+                        'tour_url' => $tourUrl,
+                    ]
+                ]
+            ],
+            'from' => [
+                'email' => $sender
+            ],
+            'template_id' => $templateId,
+        ];
+
+        if (!\App::environment('local')) {
+            $dynamicContent = array_merge($dynamicContent, [
+                'attachments' => [
+                    [
+                        'content' => base64_encode(file_get_contents($baseUrl . $model->pdf_download_path)),
+                        'type' => 'application/pdf',
+                        'filename' => 'my_museum_tour-' . $model->id . '.pdf',
+                        'disposition' => 'attachment'
+                    ],
+                ],
+            ]);
+        }
+
+        $sendgridurl = 'https://api.sendgrid.com/v3/mail/send';
```

Comment:
Should this url also be in the config, along with the api key, email, etc.?

Reply:
Yes!